sonarts-core/src/tools/new-rule.ts

```diff
@@ -0,0 +1,271 @@
+import * as fs from "fs";
+import * as stringify from "json-stable-stringify";
+
+const rootFolder = `${__dirname}/../../..`;
+const rspecRuleFolder = `${rootFolder}/sonarts-sq-plugin/sonar-typescript-plugin/src/main/resources/org/sonar/l10n/typescript/rules/typescript`;
+
+if (process.argv.length != 4) {
+  showHelp();
+  throw new Error("invalid number of arguments");
+}
+
+let rspecId = process.argv[2];
+const ruleClassName = process.argv[3];
+
+verifyClassName(ruleClassName);
+verifyRspecId(rspecId);
+
+const ruleNameDash: string = getDashName(ruleClassName);
+const { ruleTitle, rspecKey } = getRuleTitleAndRspecKey(rspecRuleFolder, rspecId);
+
+//// From README.md:
+//- Add rule key to tslint-sonarts.json
+updateSonarTsJson(rootFolder, ruleNameDash);
+
+//- Create file for rule implementation in src/rules. File name should start with lower case and have suffix Rule
+//- Create test folder in test/rules with the name of the rule file
+//- In this folder create files <rule file name>.test.ts and <rule file name>.lint.ts
+createFiles(rootFolder, ruleClassName, ruleNameDash, ruleTitle, rspecKey);
+
+//- In folder docs/rules create rule documentation file <rule key>.md
+const ruleDescription: string = `# ${ruleNameDash}\n\n${getDescription(rspecRuleFolder, rspecId)}`;

```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Type string trivially inferred from a string literal, remove type annotation ([typescript:S3257](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3257))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=michalb%2Fnew-rules-script&issues=AWC8Oukl_Q-0dz4JAzUY&open=AWC8Oukl_Q-0dz4JAzUY)

[ISSUE_KEY]: AWC8Oukl_Q-0dz4JAzUY